File: views/src/main/java/dlgdev/weighttracker/views/entry/WeightEntryAdapter.java
package dlgdev.weighttracker.views.entry;

import android.database.Cursor;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.text.DateFormat;
import java.util.Date;

import javax.inject.Inject;

import dlgdev.weighttracker.R;

import static dlgdev.weighttracker.domain.db.WeightCheckerDatabase.WeightEntryColumns.DATE;
import static dlgdev.weighttracker.domain.db.WeightCheckerDatabase.WeightEntryColumns.WEIGHT;

public class WeightEntryAdapter extends RecyclerView.Adapter<WeightEntryViewHolder> {
	Cursor items;
	DateFormat dateFormat;

	@Inject public WeightEntryAdapter(DateFormat dateFormat) {
		this.dateFormat = dateFormat;
	}

	@Override public WeightEntryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		View v = LayoutInflater.from(parent.getContext())
				.inflate(R.layout.weight_entry_view, parent, false);
		return new WeightEntryViewHolder(v);
	}

	@Override public void onBindViewHolder(WeightEntryViewHolder holder, int position) {
		if (items.moveToPosition(position)) {
			String date = dateFormat.format(new Date(items.getLong(items.getColumnIndex(DATE))));
			String entry = items.getString(items.getColumnIndex(WEIGHT));
			holder.setEntry(date, entry);
		}
	}

	@Override public int getItemCount() {
		return items.getCount();
	}

	public void setData(Cursor data) {
		this.items = data;
		this.notifyDataSetChanged();
	}
}

File: views/src/main/java/dlgdev/weighttracker/views/entry/WeightEntryAdapter.kt
package dlgdev.weighttracker.views.entry

import android.database.Cursor
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import dlgdev.weighttracker.R
import dlgdev.weighttracker.domain.db.WeightCheckerDatabase.WeightEntryColumns.DATE
import dlgdev.weighttracker.domain.db.WeightCheckerDatabase.WeightEntryColumns.WEIGHT
import java.text.DateFormat
import java.util.*
import javax.inject.Inject

class WeightEntryAdapter @Inject constructor(internal var dateFormat: DateFormat) : RecyclerView.Adapter<WeightEntryViewHolder>() {
    var items: Cursor? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeightEntryViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.weight_entry_view, parent, false)
        return WeightEntryViewHolder(v)
    }

    override fun onBindViewHolder(holder: WeightEntryViewHolder, position: Int) {
        if (items != null && items!!.moveToPosition(position)) {
            val date = dateFormat.format(Date(items!!.getLong(items!!.getColumnIndex(DATE))))
            val entry = items!!.getString(items!!.getColumnIndex(WEIGHT))
            holder.setEntry(date, entry)
        }
    }

    override fun getItemCount(): Int {
        return if(items != null) items!!.count else 0
    }

    fun setData(data: Cursor) {
        this.items = data
        this.notifyDataSetChanged()
    }
}
